### PR TITLE
Add app code checker for CI

### PR DESCRIPTION
**Pull request by ChristophWurst**
Adding this for the sake of completeness. Though we hopefully drop this at some point.

Rendered:

Bildschirmfoto von 2020-11-26 11-36-17

### CODE REVIEW

File: developer_manual/digging_deeper/continuous_integration.rst [hunk 1]
```diff
@@ -65,6 +65,45 @@ A lint of all php source files can find syntax errors that could crash the appli
...
+      name: Nextcloud ${{ matrix.nextcloud-versions }} app code check
+      steps:
+      - name: Set up php7.4
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: 7.4
+          extensions: ctype,curl,dom,gd,iconv,intl,json,mbstring,openssl,posix,sqlite,xml,zip
+          coverage: xdebug
+      - name: Checkout Nextcloud
+        run: git clone https://github.com/nextcloud/server.git --recursive --depth 1 -b ${{ matrix.nextcloud-versions }} nextcloud
```
Comment (skjnldsv) [inline developer_manual/digging_deeper/continuous_integration.rst:93]: Why ot use the `actions/checkout@master` action?

File: developer_manual/digging_deeper/continuous_integration.rst [hunk 2]
```diff
@@ -65,6 +65,45 @@ A lint of all php source files can find syntax errors that could crash the appli
...
+      - name: Run tests
+        run: php -f nextcloud/occ maintenance:install --database-name oc_autotest --database-user oc_autotest --admin-user admin --admin-pass admin --database sqlite --database-pass=''
+      - name: Checkout
+        uses: actions/checkout@master
```
Comment (skjnldsv) [inline developer_manual/digging_deeper/continuous_integration.rst:97]: We should use an official tag, not master 
`actions/checkout@v2`

File: developer_manual/digging_deeper/continuous_integration.rst (hunk 1 above)
Comment (skjnldsv) [inline developer_manual/digging_deeper/continuous_integration.rst:93]:
```suggestion
      - name: Checkout Nextcloud
        uses: actions/checkout@v2
        with:
          repository: nextcloud/server
          ref:  ${{ matrix.nextcloud-versions }}
```

**Comment by nickvergessen** [comment]
app:check-code should be deprecated and not be used anymore.
Psalm has you covered and is much more accurate. The lists for the check-code command have not been updated in quite a while (oldest deprecation is 14: https://github.com/nextcloud/server/blob/5bf3d1bb384da56adbf205752be8f840aac3b0c5/lib/private/App/CodeChecker/DeprecationCheck.php )

File: developer_manual/digging_deeper/continuous_integration.rst (hunk 1 above)
Comment (ChristophWurst) [reply developer_manual/digging_deeper/continuous_integration.rst:93]: :shrug: I copied it from Mail. IIRC there was a time when the checkout action didn't work for us

**Comment by ChristophWurst** [comment]
> app:check-code should be deprecated and not be used anymore.
> Psalm has you covered and is much more accurate. The lists for the check-code command have not been updated in quite a while (oldest deprecation is 14: https://github.com/nextcloud/server/blob/5bf3d1bb384da56adbf205752be8f840aac3b0c5/lib/private/App/CodeChecker/DeprecationCheck.php )

Well then there is really no point in continuing this.

**Comment by MorrisJobke** [comment]
> The lists for the check-code command have not been updated in quite a while (oldest deprecation is 14:

But we can add it them again?

Can Psalm also check for deprecations? The private namespace usage is most likely covered by the usage of @ChristophWurst library that provides the OCP interfaces only (instead of using the server code itself).

**Comment by ChristophWurst** [comment]
> Can Psalm also check for deprecations?

Yes: https://psalm.dev/docs/running_psalm/issues/DeprecatedMethod/

> The private namespace usage is most likely covered by the usage of @ChristophWurst library that provides the OCP interfaces only

Also yes :wink:

**Pull request closed without merging**